cli/ssh.go

```diff
@@ -183,7 +183,9 @@ func (r *RootCmd) ssh() *serpent.Command {
 
 			namedWorkspace := strings.TrimPrefix(inv.Args[0], hostPrefix)
 			// Support "--" as a delimiter between owner and workspace name
-			namedWorkspace = strings.ReplaceAll(namedWorkspace, "--", "/")
+			namedWorkspace = strings.Replace(namedWorkspace, "--", "/", 1)
+			// Support "--" as a delimiter between workspace name and agent
+			namedWorkspace = strings.Replace(namedWorkspace, "--", ".", 1)
```

Comment:
Works for me.
Is there more tweaking you want to do before I start merging things?

Reply:
Ah I see https://github.com/coder/vscode-coder/pull/420 in progress 😎

Reply:
Ready to merge, thanks! Just made this last-minute tweak to support the VS Code extension shifting over to `coder ssh`: https://github.com/coder/vscode-coder/pull/420. That PR is the last I was planning for this series.

Reply:
@aaronlehmann check out the change I made to the comment above when you get a second :)

Reply:
This basically exists to support the VS Code extension, which formats hosts as `<owner>--<workspace>--<agent>`. I was hoping to avoid changing its host format, since the extension needs to remain backward-compatible with older versions of the CLI and use `vscodessh` in those cases.

The unambiguous way to specify an agent would be to use `.` directly.

Reply:
Hmmm. Can we be more strict about `--`'s usage then? Maybe require a fully qualified host with exactly two `--`s in it before modifying it?

Reply:
The JetBrains plugin will use hosts formatted like `<owner>--<workspace>.<agent>`. Would you like me to update the JB plugin to use `<owner>--<workspace>--<agent>` for consistency with VS Code?

Alternatively, we could make the VS Code host name format conditional on whether it's using `ssh` or `vscodessh`, and avoid this second `strings.Replace`.

Reply:
Hmm, actually it looks like using `.` in the VS Code host will work even with `vscodessh`, because `--<agent>` is optional. So I can make that change unconditionally, and remove the hack here. Let me do that...

Reply:
Context: Not a hill I plan to die on haha. I can just imagine finding `<workspace>/<agent>` in a log difficult to debug or search for.

Reply:
Huh. I wonder why those were different? Yeah if owner--workspace.agent is backwards compatible with vscodessh let's do that, then they'll be consistent across the two plugins as well.

Reply:
Updated.